tests/php/includes/class-wc-cart-test.php

```diff
@@ -61,6 +61,41 @@ public function test_add_variation_to_the_cart_with_empty_attributes() {
 		$product->delete( true );
 	}
 
+	/**
+	 * @testdox should throw a notice to the cart if using variation_id
+	 * that doesn't belong to specified variable product.
+	 */
+	public function test_add_variation_to_the_cart_invalid_variation_id() {
+		WC()->cart->empty_cart();
+		WC()->session->set( 'wc_notices', null );
+
+		$variable_product = WC_Helper_Product::create_variation_product();
+		$single_product   = WC_Helper_Product::create_simple_product();
+
+		// Add variation using parent id.
+		WC()->cart->add_to_cart(
+			$variable_product->get_id(),
+			1,
+			$single_product->get_id()
+		);
+		$notices = WC()->session->get( 'wc_notices', array() );
+
+		// Check that the second add to cart call increases the quantity of the existing cart-item.
```

Comment:
:thinking: Should we re-word? I might be misreading, but I think we only have _one_ `add_to_cart()` call in this test and there is no existing cart item (because we've emptied the cart, and because the above code shouldn't result in anything being added to the cart).

Reply:
Sorry for my copypasta, I missed this one. Thanks.